src/DataStructures/Tensor/TensorData.hpp

```diff
@@ -0,0 +1,63 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <iosfwd>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief An untyped tensor component with a name for observation.
+ *
+ * The name should be a path inside an H5 file, typically starting with the name
+ * of the volume file. For example,
+ * `element_volume_data.vol/[ElementIdName]/psi_xx`.
```

Comment:
This may become obvious once we actually start producing output, but are the "H5 file" and the "volume file" the same thing?